docs/Lexicon.md

```diff
@@ -2,16 +2,58 @@
 
 ### Definitions
 
+### A
+
+* ANN
+  Artificial Neural Network.  A computational framework based on a synthetic
+  construction of a the biological neural network in brains.
+
 #### C
 
+* CNN
+  Convolutional Neural Network.  A subtype of DNNs, using feed-forwarding.  They
+  are most commonly applied to analysis of images.
+
 * CVP
   Constant Value Propagation
 
 ### D
 
+* DAG
+  Directed Acylic Graph
+
 * DNM
   Do Not Merge
 
+* DNN
+  Deep Neural Networks.  A neural network with multiple layers between the input
+  and output layers.  These work well with linear and non-linear relationships.
+
+### G
+
+* GEMM
+  Generialied Matrix Multiply.
+
+* GRU
+  Gated Recurrent Unit.  A gating mechanism for neural networks.  They are
+  similar to LSTM but exhibit better performance characteristics on smaller data
+  sets.
+
+### L
+
+* LSTM
+  Long Short Term Memory.  Units of RNNs consisting of a cell, input gate,
```

Comment:
there are quite a few double spacing here and there in the file.

Reply:
2spaces4life